AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.m

```diff
@@ -147,22 +149,24 @@ + (void)saveToken:(MSTokenResult *)tokenResult {
  * Get back the Cached Cosmos DB token
  * If the DB token has expired, that token is deleted from KeyChain
  */
-+ (MSTokenResult *_Nullable)retrieveCachedToken:(NSString *)partitionName {
-  if (partitionName) {
-    NSString *tokenString = [MSKeychainUtil stringForKey:[MSTokenExchange tokenKeyNameForPartition:partitionName]];
++ (MSTokenResult *_Nullable)retrieveCachedToken:(NSString *)partition expiredTokenIncluded:(BOOL)expiredTokenIncluded {
+  if (partition) {
+    NSString *tokenString = [MSKeychainUtil stringForKey:[MSTokenExchange tokenKeyNameForPartition:partition]];
     if (tokenString) {
       MSTokenResult *tokenResult = [[MSTokenResult alloc] initWithString:tokenString];
       NSDate *currentUTCDate = [NSDate date];
       NSDate *tokenExpireDate = [MSUtility dateFromISO8601:tokenResult.expiresOn];
       if ([currentUTCDate laterDate:tokenExpireDate] == currentUTCDate) {
-        [MSTokenExchange removeCachedToken:partitionName];
-        MSLogWarning([MSDataStore logTag], @"The token in the cache has expired for the partition : %@.", partitionName);
+        MSLogWarning([MSDataStore logTag], @"The token in the cache has expired for the partition : %@.", partition);
+        if (expiredTokenIncluded) {
+          return tokenResult;
```

Comment:
I think this line is uncovered

Reply:
I think it should be - but I think that this is unrelated to this PR